compiler/rustc_borrowck/src/region_infer/mod.rs

```diff
@@ -2236,6 +2236,29 @@ impl<'tcx> RegionInferenceContext<'tcx> {
     pub(crate) fn universe_info(&self, universe: ty::UniverseIndex) -> UniverseInfo<'tcx> {
         self.universe_causes[&universe].clone()
     }
+
+    /// Tries to find the terminator of the loop in which the region 'r' resides.
+    /// Returns the location of the terminator if found.
+    pub(crate) fn find_loop_terminator_location(
+        &self,
+        r: RegionVid,
+        body: &Body<'_>,
+    ) -> Option<Location> {
+        let scc = self.constraint_sccs.scc(r.to_region_vid());
+        let elements = self.scc_values.locations_outlived_by(scc).map(RegionElement::Location);
+        for element in elements {
+            if let RegionElement::Location(loc) = element {
+                let bb = &body[loc.block];
+                if let Some(terminator) = &bb.terminator {
+                    // terminator of a loop should be TerminatorKind::FalseUnwind
+                    if let TerminatorKind::FalseUnwind { .. } = terminator.kind {
+                        return Some(Location { block: loc.block, statement_index: 0 });
```

Comment:
Why not the terminator location?

Reply:
Sure, just use loc with `return Some(loc);' will be simpler.